stan/math/prim/meta/is_eigen.hpp

```diff
@@ -43,14 +73,20 @@ struct is_eigen<
     : std::true_type {};
 
 namespace internal {
-template <typename T>
+template <typename T, typename Enable = void>
```

Comment:
This new template parameter needs doc.  If it's called Enable, why set it to void by default?  Shouldn't it be boolean?

Reply:
Added doc. `Enable` is set to void here since when the other enable_if specializations fail the base class needs to be able to have all of it's template parameters resolved. T will be resolved by the user calling `is_eigen<Type>` but the Enable needs to also have a valid type to work.

What's the reason behind using boolean?